### PR TITLE
[Merged by Bors] - feat(Dynamics/Ergodic): ergodicity of `(a * ·)`

### PR DESCRIPTION
**Pull request by urkud**
---

- [x] depends on: #16576 
- [x] depends on: #16281

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Dynamics/Ergodic/Action/OfMinimal.lean [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+/-!
+# Ergodicity from minimality
+
+
+-/
+
+open MeasureTheory Filter Set Function
+open scoped Pointwise
+
+variable {X : Type*} [TopologicalSpace X] [R1Space X] [MeasurableSpace X] [BorelSpace X]
```
Comment (urkud) [inline Mathlib/Dynamics/Ergodic/Action/OfMinimal.lean:19]: @sgouezel Could you please have a look at the theorems in this file? Do I use the right assumptions, or some of them aren't needed?

I'm going to add docstrings and move more assumptions to `variable`s once API is stable.

Comment (urkud) [reply Mathlib/Dynamics/Ergodic/Action/OfMinimal.lean]: I reorganized to use `section`s and `variable`s. I'll add more docstrings tomorrow.

File: Mathlib/Dynamics/Ergodic/Action/OfMinimal.lean [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+    MulAction.mem_aestabilizer, ← preimage_smul, hs]
+
+ end MulActionGroup
+
+section TopologicalGroup
+
+variable {G : Type*} [Group G] [TopologicalSpace G] [TopologicalGroup G] [MeasurableSpace G]
+
+/-- If the left multiplication by `g` is ergodic
+with respect to a measure which is open on nonempty open sets,
```
Comment (sgouezel) [inline Mathlib/Dynamics/Ergodic/Action/OfMinimal.lean:147]:
```suggestion
with respect to a measure which is positive on nonempty open sets,
```

File: Mathlib/Dynamics/Ergodic/Action/OfMinimal.lean [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+then the integer powers of `g` are dense in `G`. -/
+@[to_additive "If the left addition of `g` is ergodic
+with respect to a measure which is open on nonempty open sets,
```
Comment (sgouezel) [inline Mathlib/Dynamics/Ergodic/Action/OfMinimal.lean:150]:
```suggestion
with respect to a measure which is positive on nonempty open sets,
```

File: Mathlib/Dynamics/Ergodic/Action/OfMinimal.lean [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+The proof of the more difficult "if minimal, then ergodic" implication
+is based on the ergodicity of the left action of a group on itself
+and the following fact that we prove in `ergodic_smul_of_denseRange_pow` below:
+
+If a monoid `M` continuously acts on an R₁ topological space `X`,
+`g` is an element of `M such that its natural powers are dense in `M`,
+and `μ` is a finite inner regular measure on `X` which is ergodic with respect to the action of `M`,
+then the scalar multiplication by `g` is an ergodic map.
+
+We also prove that a continuous monoid homomorphism `f : G →* G` is dense,
```
Comment (sgouezel) [inline Mathlib/Dynamics/Ergodic/Action/OfMinimal.lean:26]:
```suggestion
We also prove that a continuous monoid homomorphism `f : G →* G` is ergodic,
```

File: Mathlib/Dynamics/Ergodic/Action/OfMinimal.lean [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+
+@[to_additive]
+theorem ergodic_mul_left_of_denseRange_zpow (hg : DenseRange (g ^ · : ℤ → G))
+    (μ : Measure G) [IsFiniteMeasure μ] [μ.InnerRegular] [μ.IsMulLeftInvariant] :
+    Ergodic (g * ·) μ :=
+  ergodic_smul_of_denseRange_zpow hg μ
+
+@[to_additive]
+theorem ergodic_mul_left_iff_denseRange_zpow (μ : Measure G) [IsFiniteMeasure μ]
+    [μ.InnerRegular] [μ.IsMulLeftInvariant] [μ.IsOpenPosMeasure] :
```
Comment (sgouezel) [inline Mathlib/Dynamics/Ergodic/Action/OfMinimal.lean:198]: I agree that these assumptions are natural, but isn't the theorem true under the weaker assumption where you replace `IsOpenPosMeasure` with `NeZero`?

Comment (urkud) [reply Mathlib/Dynamics/Ergodic/Action/OfMinimal.lean:198]: How do you prove it with `NeZero`?

Comment (urkud) [reply Mathlib/Dynamics/Ergodic/Action/OfMinimal.lean]: UPD: in this context, our typeclass search can prove `IsOpenPosMeasure μ` from `NeZero μ`.

**Comment by sgouezel** [comment]
bors r+
Thanks!

**Pull request closed without merging**